components/dashboard/src/components/QuickStart.tsx

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2024 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { FC, useEffect, useState } from "react";
+import { useAuthProviderDescriptions } from "../data/auth-providers/auth-provider-descriptions-query";
+import { parseError, redirectToAuthorize, redirectToOIDC } from "../provider-utils";
+import { useCurrentUser } from "../user-context";
+import { useHistory } from "react-router";
+import { AppLoading } from "../app/AppLoading";
+import { Link } from "react-router-dom";
+
+const parseErrorFromSearch = (search: string): string => {
+    const searchParams = new URLSearchParams(search);
+    const message = searchParams.get("message");
+    if (message?.startsWith("error:")) {
+        const parsed = parseError(message);
+        return typeof parsed === "string" ? parsed : `${parsed.error}: ${parsed.description}`;
+    }
+
+    return "";
+};
+
+const QuickStart: FC = () => {
+    const [error, setError] = useState(parseErrorFromSearch(window.location.search));
+    const { data: authProviders, isLoading: authProvidersLoading } = useAuthProviderDescriptions();
+    const user = useCurrentUser();
+    const history = useHistory();
+
+    useEffect(() => {
+        if (authProvidersLoading || error) {
+            return;
+        }
+
+        const hash = window.location.hash.slice(1);
+        let contextUrl: URL;
+        try {
+            contextUrl = new URL(hash);
+        } catch {
+            setError("Invalid context URL");
+            return;
+        }
+        const relevantAuthProvider = authProviders?.find((provider) => provider.host === contextUrl.host);
+
+        if (!user) {
+            if (relevantAuthProvider) {
+                void redirectToAuthorize({
+                    login: true,
+                    host: relevantAuthProvider.host,
+                    overrideScopes: true,
+                });
+
+                return;
+            }
+            void redirectToOIDC({
```

Comment:
In both these functions we should provide the redirect URL here instead of constructing it in the function itself.